packages/pangea-sdk/pangea/verify_audit.py

```diff
@@ -78,8 +97,31 @@ def log_section(msg: str):
 formatter = VerifierLogFormatter()
 
 
-def _verify_hash(data: Dict, data_hash: str) -> Optional[bool]:
+def get_pangea_roots(tree_name: str, tree_sizes: Iterable[int]) -> Dict[int, Root]:
+    ans: Dict[int, Root] = {}
+    if audit is None:
+        return ans
+
+    for size in tree_sizes:
+        try:
+            resp = audit.root(size)
+
+            if resp.status != "Success":
+                raise ValueError(resp.status)
+            elif resp.result is None or resp.result.data is None:
+                raise ValueError("No result")
+            elif resp.result.data.tree_name != tree_name:
+                raise ValueError("Invalid Pangea Token provided")
+
+            ans[int(size)] = resp.result.data
+        except Exception as e:
+            logger.error(f"Error fetching root from Pangea for size {size}: {str(e)}")
```

Comment:
This case could be due to a wrong token, if you are trying to get root 100 and current token is associated to a smaller tree we'll get an error. Should we process those cases?

Reply:
True. I will catch that case and print "Invalit Pangea Token provided"